docs/zkapps/faq.mdx

```diff
@@ -4,21 +4,20 @@ title: FAQ
 
 ## Does SnarkyJS compile my JavaScript code to an arithmetic circuit?
 
-No, SnarkyJS **does NOT compile JS into anything else**! This is a very common misconception about how SnarkyJS works. It may be caused by the emphasis on compilation in other zk ecosystems -- many of which have custom languages for writing zk circuits.
+No, SnarkyJS **does NOT compile into anything else**! In contrast to other zk ecosystems, SnarkyJS is just a JS library. It creates zk circuits from user code by _executing_ that code. If you have a smart contract with a `@method myMethod()`, for example, we will simply call `myMethod();` during proof generation.
 
-SnarkyJS, by contrast, is just a JavaScript library. It creates zk circuits from user code by _executing_ that code. For example, if you have a SmartContract with a `@method myMethod()`, during proof generation we will simply call that method: `myMethod();`
+This works because SnarkyJS sets up some global state - a "circuit" - where it collects variables and constraints. The use of functions like `Field.mul` or `Bool.assertEquals` inside your smart contract methods will add corresponding variables and constraints to the global circuit.
 
-This works because SnarkyJS sets up some global state - a "circuit" - where it collects variables and constraints, and functions like `Field.mul` or `Bool.assertEquals`, which you use inside your method, add variables/constraints to that global circuit.
+This has some implications:
 
-This has some implications for how your code behaves, and what part of it is included in the zk proof:
+- In order to turn your logic into a proof, you'll need to use SnarkyJS built-in datatypes such as `Field`. You'll also need to use the SnarkyJS functions that operate on them, like `Field.mul()`.
+  - A statement like `x.mul(y)` will add a generic PLONK gate to your circuit. It will also return a variable that you can use in further statements, which will get wired to the multiplication gate.
+  - Some SnarkyJS functions allow you to turn normal JS datatypes into `Field` elements and back, such as `Field.toString()`. These functions don't add anything to your circuit. They will typically clarify this in a doc-comment.
+- Conventional JS code such as `'hello world'.split('').join(' ')` that doesn't use SnarkyJS built-ins will not be included in your zk proof in any way since it will not add anything to your circuit.
+  - Why? Because it doesn't call any of the functions that build the circuit.
+  - There's nothing wrong with having circuit-less code like the above inside your method, as long as you're aware of what it's (not) doing.
```

Comment:
> circuit-less

not sure if I like "circuit-less code" here better than "circuit-unrelated code" 🤔 maybe "non-circuit code"?